doc/iterations/2025.6/add_impact_analysis_before_tool_execution.adoc

```diff
@@ -0,0 +1,56 @@
+= Add Impact Analysis Before Tool Execution in Trees
+
+== Problem
+
+Sirius Web allows specifier to add an impact analysis on tools executed on diagrams, but not on tools in contextual menus of trees.
+As in diagrams, some tool executions from tree contextual menus can have a significant impact on the model, and the specifier may want to warn the end user of this impact before validating the execution.
+
+== Key Result
+
+The specifier can declare that an impact analysis will be triggered before the execution of a tool on a tree.
+When a user triggers this tool, they receive initial feedback on the changes that will occur after the tool execution.
+The user can then decide whether to apply the tool or not.
+
+=== Acceptance criteria
+
+- In Studio, some impact analysis should be available before the execution of the tool that updates an imported library.
```

Comment:
For the record, my plan is have separate PRs for the addition of impact analysis in trees and the computation of the impact analysis for the update of libraries.
The second part looks quite complex on its own, and may require its own shape.
If it is the case, I may deliver an initial implementation of the impact analysis with a "simpler" example (e.g. the _toggle abstract_ tool in the studio). Let me know if it works for you

Reply:
In my opinion, the very first step should be the extraction of the common parts to be able to have a very basic prototype, even a completely hardcoded one, which could show an impact analysis dialog from the explorer. Without even considering changes in the tree meta-model or things like that. Some hardcoded tree item context menu entry for flow for example. It doesn't even have to call a mutation in the backend, we can even fake that, that's a detail.

Once you have successfully extracted what can be reused from the diagram part and you can reuse it in the tree part then you can start working on properly supporting it for trees.